agent/log_streamer.go

```diff
@@ -97,14 +107,29 @@ func (ls *LogStreamer) Process(output []byte) error {
 	ls.processMutex.Lock()
 	defer ls.processMutex.Unlock()
 
+	warnedAboutSize := false
+
 	for len(output) > 0 {
+		// Have we exceeded the max size?
+		// (This check is also performed on the server side.)
+		if ls.bytes > ls.conf.MaxSizeBytes && !warnedAboutSize {
+			ls.logger.Warn("The job log has reached %s in size, which has "+
+				"exceeded the maximum size (%s). Further logs may be dropped "+
+				"by the server, and a future version of the agent will stop "+
+				"sending logs at this point.",
+				humanize.Bytes(ls.bytes), humanize.Bytes(ls.conf.MaxSizeBytes))
```

Comment:
@matthewborden suggested this should just be a warning for now. Here's a cut of that.